Content.Client/Inventory/ClientInventorySystem.cs

```diff
@@ -85,23 +86,27 @@ public override void Initialize()
 
         private void OnDidUnequip(EntityUid uid, ClientInventoryComponent component, DidUnequipEvent args)
         {
-            if (component.SlotButtons.TryGetValue(args.Slot, out var buttons))
-            {
-                foreach (var button in buttons)
-                {
-                    _itemSlotManager.SetItemSlot(button, null);
-                }
-            }
+            UpdateComponentUISlot(component, args.Slot, null);
         }
 
         private void OnDidEquip(EntityUid uid, ClientInventoryComponent component, DidEquipEvent args)
         {
-            if (component.SlotButtons.TryGetValue(args.Slot, out var buttons))
+            UpdateComponentUISlot(component, args.Slot, args.Equipment);
+        }
+
+        private void UpdateComponentUISlot(ClientInventoryComponent component, string slot, EntityUid? entity)
+        {
+            if (!component.SlotButtons.TryGetValue(slot, out var buttons))
+                return;
+
+            UpdateUISlot(buttons, entity);
+        }
```

Comment:
@PaulRitter I'm not sure if doing the resolve pattern here makes sense since the EntityUid is nullable.
```suggestion
        private void UpdateComponentUISlot(EntityUid? entity, string slot, ClientInventoryComponent? component = null)
        {
            if (entity != null && !Resolve(entity.Value, ref component))
                return;

            if (component == null || !component.SlotButtons.TryGetValue(slot, out var buttons))
                return;

            UpdateUISlot(buttons, entity);
        }
```

Reply:
why do you need to make the entityuid nullable?

Reply:
ItemSlotManager.SetItemSlot takes in the ItemSlotButton and a nullable EntityUid. Null means that there is no entity in the button/ContainerSlot.